addons/account/models/account_move_line.py

```diff
@@ -915,6 +915,7 @@ def _compute_tax_key(self):
                     'tax_tag_ids': [(6, 0, line.tax_tag_ids.ids)],
                     'partner_id': line.partner_id.id,
                     'move_id': line.move_id.id,
+                    'display_type': line.display_type,
```

Comment:
If you are adding it in here, you must also add it in the keys of `compute_all_tax`, otherwise it will burn through ids instead of recycling them (look at how `to_delete` and `to_create` are merged as a `write` in `_sync_dynamic_line`)

Reply:
@william-andre  I followed your advice and that lead me to patch a test as well as it caused one of the invoice lines to split into two lines. I think it should be equivalent as it balances to the same amounts but let me know if you think that's an issue.

<details>
  <summary>The details for the two new lines if needed</summary>
[{'id': 138, 'analytic_distribution': False, 'analytic_precision': 0, 'move_id': (45, 'Draft Invoice INV/2019/00001'), 'journal_id': (240, 'Customer Invoices'), 'company_id': (35, 'company_1_data'), 'company_currency_id': (2, 'USD'), 'move_name': 'INV/2019/00001', 'parent_state': 'draft', 'date': datetime.date(2019, 1, 1), 'ref': False, 'is_storno': False, 'sequence': 10000, 'move_type': 'out_invoice', 'account_id': (1572, '251000 Tax Received'), 'name': 'Tax 15%', 'debit': 0.0, 'credit': 150.0, 'balance': -150.0, 'cumulated_balance': 0.0, 'currency_rate': 1.0, 'amount_currency': -150.0, 'currency_id': (2, 'USD'), 'is_same_currency': True, 'partner_id': (124, 'partner_a'), 'reconcile_model_id': False, 'payment_id': False, 'statement_line_id': False, 'statement_id': False, 'tax_ids': [], 'group_tax_id': False, 'tax_line_id': (168, 'Tax 15%'), 'tax_group_id': (2, 'Tax 15%'), 'tax_base_amount': 1000.0, 'tax_repartition_line_id': (702, 'account.tax.repartition.line,702'), 'tax_tag_ids': [], 'tax_audit': '', 'tax_tag_invert': True, 'amount_residual': 0.0, 'amount_residual_currency': 0.0, 'reconciled': False, 'full_reconcile_id': False, 'matched_debit_ids': [], 'matched_credit_ids': [], 'matching_number': False, 'is_account_reconcile': False, 'account_type': 'liability_current', 'account_internal_group': 'liability', 'account_root_id': (50053, '25'), 'display_type': 'tax', 'product_id': False, 'product_uom_id': False, 'product_uom_category_id': False, 'quantity': 0.0, 'date_maturity': False, 'price_unit': 0.0, 'price_subtotal': 0.0, 'price_total': 0.0, 'discount': 0.0, 'term_key': False, 'tax_key': {'tax_repartition_line_id': 702, 'group_tax_id': False, 'account_id': 1572, 'currency_id': 2, 'analytic_distribution': False, 'tax_ids': [(6, 0, [])], 'tax_tag_ids': [(6, 0, [])], 'partner_id': 124, 'move_id': 45, 'display_type': 'tax'}, 'compute_all_tax': {}, 'compute_all_tax_dirty': False, 'epd_key': False, 'epd_needed': False, 'epd_dirty': True, 'analytic_line_ids': [], 'discount_date': False, 'discount_amount_currency': 0.0, 'discount_balance': 0.0, 'discount_percentage': 0.0, 'blocked': False, 'is_refund': False, '__last_update': datetime.datetime(2023, 4, 5, 8, 52, 6, 842107), 'display_name': 'INV/2019/00001 Tax 15%', 'create_uid': (24, 'Because I am accountman!'), 'create_date': datetime.datetime(2023, 4, 5, 8, 52, 6, 842107), 'write_uid': (24, 'Because I am accountman!'), 'write_date': datetime.datetime(2023, 4, 5, 8, 52, 6, 842107), 'move_attachment_ids': [], 'expected_pay_date': False, 'is_downpayment': False, 'sale_line_ids': [], 'asset_ids': [], 'non_deductible_tax_value': 0.0, 'followup_line_id': False, 'last_followup_date': False, 'next_action_date': False, 'invoice_date': datetime.date(2019, 1, 1), 'invoice_origin': False}]

---

[{'id': 140, 'analytic_distribution': False, 'analytic_precision': 0, 'move_id': (45, 'Draft Invoice INV/2019/00001'), 'journal_id': (240, 'Customer Invoices'), 'company_id': (35, 'company_1_data'), 'company_currency_id': (2, 'USD'), 'move_name': 'INV/2019/00001', 'parent_state': 'draft', 'date': datetime.date(2019, 1, 1), 'ref': False, 'is_storno': False, 'sequence': 10000, 'move_type': 'out_invoice', 'account_id': (1572, '251000 Tax Received'), 'name': 'Tax 15%', 'debit': 11.25, 'credit': 0.0, 'balance': 11.25, 'cumulated_balance': 0.0, 'currency_rate': 1.0, 'amount_currency': 11.25, 'currency_id': (2, 'USD'), 'is_same_currency': True, 'partner_id': (124, 'partner_a'), 'reconcile_model_id': False, 'payment_id': False, 'statement_line_id': False, 'statement_id': False, 'tax_ids': [], 'group_tax_id': False, 'tax_line_id': (168, 'Tax 15%'), 'tax_group_id': (2, 'Tax 15%'), 'tax_base_amount': -75.0, 'tax_repartition_line_id': (702, 'account.tax.repartition.line,702'), 'tax_tag_ids': [], 'tax_audit': '', 'tax_tag_invert': True, 'amount_residual': 0.0, 'amount_residual_currency': 0.0, 'reconciled': False, 'full_reconcile_id': False, 'matched_debit_ids': [], 'matched_credit_ids': [], 'matching_number': False, 'is_account_reconcile': False, 'account_type': 'liability_current', 'account_internal_group': 'liability', 'account_root_id': (50053, '25'), 'display_type': 'tax', 'product_id': False, 'product_uom_id': False, 'product_uom_category_id': False, 'quantity': 0.0, 'date_maturity': False, 'price_unit': 0.0, 'price_subtotal': 0.0, 'price_total': 0.0, 'discount': 0.0, 'term_key': False, 'tax_key': {'tax_repartition_line_id': 702, 'group_tax_id': False, 'account_id': 1572, 'currency_id': 2, 'analytic_distribution': False, 'tax_ids': [(6, 0, [])], 'tax_tag_ids': [(6, 0, [])], 'partner_id': 124, 'move_id': 45, 'display_type': 'tax'}, 'compute_all_tax': {}, 'compute_all_tax_dirty': False, 'epd_key': False, 'epd_needed': False, 'epd_dirty': True, 'analytic_line_ids': [], 'discount_date': False, 'discount_amount_currency': 0.0, 'discount_balance': 0.0, 'discount_percentage': 0.0, 'blocked': False, 'is_refund': False, '__last_update':    datetime.datetime(2023, 4, 5, 8, 52, 6, 842107), 'display_name': 'INV/2019/00001 Tax 15%', 'create_uid': (24, 'Because I am accountman!'), 'create_date': datetime.datetime(2023, 4, 5, 8, 52, 6, 842107), 'write_uid': (24, 'Because I am accountman!'), 'write_date': datetime.datetime(2023, 4, 5, 8, 52, 6, 842107), 'move_attachment_ids': [], 'expected_pay_date': False, 'is_downpayment': False, 'sale_line_ids': [], 'asset_ids': [], 'non_deductible_tax_value': 0.0, 'followup_line_id': False, 'last_followup_date': False, 'next_action_date': False, 'invoice_date': datetime.date(2019, 1, 1), 'invoice_origin': False}]

</details>